### PR TITLE
Add Ingress-NGINX Controller migration document

### PR DESCRIPTION
**Pull request by ADubhlaoich**
### Proposed changes

This commit is a close to 1:1 conversion of a blog post released on the main NGINX website. It makes certain blocks of text somewhat more concise and also cleans up the phrasing of certain things.

https://www.nginx.com/blog/migrating-from-community-ingress-controller-to-f5-nginx-ingress-controller/

It also adds some small improvements to a currently hidden document comparing the featuresets of the two Ingress Controllers: it will not be published until a later date when it can be assessed for updates, as we do not wish to misrepresent what updates have been made to the community Controller since that document was originally written.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [ ] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by vepatel — APPROVED** [review]

File: docs/content/installation/migration/ingress-nginx.md
Comment (jputrino) [inline docs/content/installation/migration/ingress-nginx.md]:
```suggestion
This document provides instructions for migrating from the community-maintained [Ingress-NGINX Controller](https://github.com/kubernetes/ingress-nginx) project to the NGINX Ingress Controller. You can do so by using NGINX's Ingress Resources or Kubernetes' built-in Ingress Resources. 

Migrating to NGINX's ingress controller implementation makes it possible to take advantage of [NGINX Plus]({{<relref "overview/nginx-plus">}}) features that are not available in the community ingress controller. 
```

File: docs/content/installation/migration/ingress-nginx.md
Comment (jputrino) [inline docs/content/installation/migration/ingress-nginx.md:44]:
```suggestion
The following two code examples correspond to a Kubernetes Ingress Resource and an [NGINX VirtualServer Resource]({{<relref "configuration/virtualserver-and-virtualserverroute-resources">}}). Although the syntax and indentation differ, they accomplish the same basic Ingress functions, used for SSL termination and Layer 7 path-based routing.
```

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:20]:
```suggestion
This page explains two different ways to migrate from the community-maintained [Ingress-NGINX Controller](https://github.com/kubernetes/ingress-nginx) project to NGINX Ingress Controller: using NGINX's Ingress Resources or with Kubernetes's built-in Ingress Resources. This is typically because of implementation differences, and to take advantage of features such as [NGINX Plus integration]({{<relref "overview/nginx-plus">}}).
```

File: docs/content/installation/migration/ingress-nginx.md
Comment (jputrino) [inline docs/content/installation/migration/ingress-nginx.md:97]:
```suggestion
NGINX Ingress Controller exposes TCP and UDP services using [TransportServer]({{<relref "configuration/transportserver-resource">}}) and [GlobalConfiguration]({{<relref "configuration/global-configuration/globalconfiguration-resource">}}) resources. These resources provide a broad range of options for TCP/UDP and TLS Passthrough load balancing. By contrast, the community Ingress-NGINX Controller exposes TCP/UDP services by using a Kubernetes ConfigMap object.
```

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [reply docs/content/installation/migration/ingress-nginx.md:20]: We don't use articles with product names.

File: docs/content/installation/migration/ingress-nginx.md
Comment (jputrino) [inline docs/content/installation/migration/ingress-nginx.md:102]:
```suggestion
Kubernetes deployments often need to extend basic Ingress rules for advanced use cases such as canary and blue-green deployments, traffic throttling, and ingress-egress traffic manipulation. The community Ingress-NGINX Controller implements many of these using Kubernetes annotations with custom Lua extensions.
```

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:33]: Format as a bullet list to make the paths more apparent.

File: docs/content/installation/migration/ingress-nginx.md
Comment (jputrino) [inline docs/content/installation/migration/ingress-nginx.md:104]:
```suggestion
These custom Lua extensions are intended for specific NGINX Ingress resource definitions and may not be as granular as required for advanced use cases. The following examples show how to convert these annotations into NGINX Ingress Controller Resources.
```

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:38]: Use `-` for bulleted lists. That's been our standard for the most part.

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:97]:
```suggestion
NGINX Ingress Controller exposes TCP and UDP services using [TransportServer]({{<relref "configuration/transportserver-resource">}}) and [GlobalConfiguration]({{<relref "configuration/global-configuration/globalconfiguration-resource">}}) resources together. These provide a broad range of options for TCP/UDP and TLS Passthrough load balancing: the community Ingress-NGINX Controller exposes these services solely using a Kubernetes ConfigMap object.
```

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:102]:
```suggestion
Kubernetes deployments often need to extend basic Ingress rules for advanced use cases such as canary and blue-green deployments, traffic throttling, and ingress-egress traffic manipulation. The community Ingress-NGINX Controller implements many of these using Kubernetes annotations with custom Lua extensions.
```

Comment (travisamartin) [reply docs/content/installation/migration/ingress-nginx.md:102]: Use the Oxford comma with lists.

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:109]:
```suggestion
Canary and blue-green deployments allow you to push code changes to production environments without disrupting existing users. NGINX Ingress Controller runs them on the data plane: to migrate from the community Ingress-NGINX Controller, you must map the latter's annotations to [VirtualServer and VirtualServerRoute resources]({{<relref "configuration/virtualserver-and-virtualserverroute-resources">}}).
```

Comment (travisamartin) [reply docs/content/installation/migration/ingress-nginx.md:109]: The style guide says not to use "execute" -- for geopolitical reasons.

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [reply docs/content/installation/migration/ingress-nginx.md:350]: F5 style guide says not to use "wish"

image


https://docs.f5net.com/pages/viewpage.action?spaceKey=GSASKF5&title=F5+Technical+Style+Guide

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:352]:
```suggestion
This table shows how Ingress-NGINX Controller annotations map to statements in the upstream field for [VirtualServer and VirtualServerRoute resources]({{<relref "configuration/virtualserver-and-virtualserverroute-resources">}}), covering load balancing, proxy timeout, proxy buffering and connection routing for a services' ClusterIP address and port.
```

Comment (travisamartin) [reply docs/content/installation/migration/ingress-nginx.md:352]: Make sure to spell check.

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:373]:
```suggestion
NGINX Ingress Controller layer can handle mTLS authentication for end systems through the presentation of valid certificates for external connections. It accomplishes this through [Policy]({{<relref "configuration/policy-resource">}}) resources, which correspond to Ingress-NGINX Controller annotations for [client certificate authentication](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#client-certificate-authentication) and [backend certificate authentication](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#backend-certificate-authentication).
```

Comment (travisamartin) [reply docs/content/installation/migration/ingress-nginx.md:373]: Don't use articles w/ product names.

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:456]: Why? It might be helpful for a warning to explain the consequences -- could help w/ troubleshooting.

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:465]: Do these need to be formatted as in-line code or perhaps bold or italic? I know we've gone back and forth on this. We should decide.

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:479]:
```suggestion
1. To redirect HTTP (80) traffic to HTTPS (443), NGINX Ingress Controller uses native NGINX `if` conditions while Ingress-NGINX Controller uses Lua.
```

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:481]:
```suggestion
The following two snippets outline Ingress-NGINX Controller annotations that correspond to annotations for NGINX Ingress Controller with NGINX Plus.
```

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:497]:
```suggestion
NGINX Ingress Controller has additional annotations for features using NGINX Plus that have no Ingress-NGINX Controller equivalent, such as active health checks and authentication using JSON Web Tokens (JWTs).
```

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [inline docs/content/installation/migration/ingress-nginx.md:511]: Same question as above -- should these be formatted as in-line code or bold or italic?

File: docs/content/overview/controller-comparison.md
Comment (travisamartin) [inline docs/content/overview/controller-comparison.md:3]:
```suggestion
description: This document describes the key differences between the community Ingress-NGINX Controller and F5 NGINX Ingress Controller.
```

File: docs/content/installation/migration/ingress-nginx.md
Comment (Jcahilltorre) [inline docs/content/installation/migration/ingress-nginx.md:30]:
```suggestion
- A working knowledge of [Ingress Controllers]({{<relref "glossary.m,d#ingress-controller">}}).
```

File: docs/content/installation/migration/ingress-nginx.md
Comment (Jcahilltorre) [inline docs/content/installation/migration/ingress-nginx.md:30]:
```suggestion
- A working knowledge of [Ingress Controllers]({{<relref "glossary.md#ingress-controller-ingress-controller">}}).
```

**Review by travisamartin — CHANGES_REQUESTED** [review]
Looks good. A few issues with spelling and style to address before merging.

File: docs/content/installation/migration/ingress-nginx.md
Comment (ADubhlaoich) [reply docs/content/installation/migration/ingress-nginx.md:456]: I have no idea why, but I'll ask Jason Williams later - I suspect changing the spec fields breaks the resource's ability to parse other parameters.

File: docs/content/installation/migration/ingress-nginx.md
Comment (ADubhlaoich) [reply docs/content/installation/migration/ingress-nginx.md:465]: I've formatted these all as in-line code to march the original blog post.

https://www.nginx.com/blog/migrating-from-community-ingress-controller-to-f5-nginx-ingress-controller/

File: docs/content/installation/migration/ingress-nginx.md
Comment (ADubhlaoich) [reply docs/content/installation/migration/ingress-nginx.md:511]: Same answer as above!

I've formatted these all as in-line code to march the original blog post.

https://www.nginx.com/blog/migrating-from-community-ingress-controller-to-f5-nginx-ingress-controller/

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [reply docs/content/installation/migration/ingress-nginx.md:459]: A warning is the highest level of caution, intended for issues that could lead to data loss or hardware failure. We need to specify the risks to keep users informed. If the consequences are unclear, perhaps this shouldn't be labeled as a warning.

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [reply docs/content/installation/migration/ingress-nginx.md:514]: The blog post isn't a style guide. I think we decided on bold? I raised the issue in Slack.

File: docs/content/installation/migration/ingress-nginx.md
Comment (travisamartin) [reply docs/content/installation/migration/ingress-nginx.md:468]: The blog post isn't a style guide. I raised the issue in Slack so we have consensus.

Comment (travisamartin) [reply docs/content/installation/migration/ingress-nginx.md:468]: The table on Proxying and Load Balancing doesn't use in-line code for parameters, making it inconsistent with other tables in the same document. This inconsistency is hard to miss.

**Review by travisamartin — APPROVED** [review]

**Pull request merged**
